// aniki/dist/esm/kitsu/enums/index.js
/**
 *  @file This file contains all enums of AnimeKitsu and MangaKitsu method, can be exported for other purpose.
 */
/**
 * @enum
 * @description This enum contain all of manga categories that use Kitsu.app
 * @since 1.3.0
 */
var EKitsuMangaCategories;
(function (EKitsuMangaCategories) {
    EKitsuMangaCategories["VIOLENCE"] = "violence";
    EKitsuMangaCategories["PLOT_CONTINUITY"] = "plot_continuity";
    EKitsuMangaCategories["STEREOTYPES"] = "stereotypes";
    EKitsuMangaCategories["TONE_CHANGES"] = "tone_changes";
    EKitsuMangaCategories["ACTION"] = "action";
    EKitsuMangaCategories["BATTLE_ROYALE"] = "battle_royale";
    EKitsuMangaCategories["GUNFIGHTS"] = "gunfights";
    EKitsuMangaCategories["NINJA"] = "ninja";
    EKitsuMangaCategories["SPACE_BATTLES"] = "space_battles";
    EKitsuMangaCategories["SAMURAI"] = "samurai";
    EKitsuMangaCategories["ADVENTURE"] = "adventure";
    EKitsuMangaCategories["ANGST"] = "angst";
    EKitsuMangaCategories["ANIME_INFLUENCED"] = "anime_influenced";
    EKitsuMangaCategories["ANTHROPOMORPHISM"] = "anthropomorphism";
    EKitsuMangaCategories["BLACKMAIL"] = "blackmail";
    EKitsuMangaCategories["ABSURDIST_HUMOUR"] = "absurdist_humour";
    EKitsuMangaCategories["BREAKING_THE_FOURTH_WALL"] = "breaking_the_fourth_wall";
    EKitsuMangaCategories["PARODY"] = "parody";
    EKitsuMangaCategories["SATIRE"] = "satire";
    EKitsuMangaCategories["VIOLENT_RETRIBUTION_FOR_ACCIDENTAL_INFRINGEMENT"] = "violent_retribution_for_accidental_infringement";
    EKitsuMangaCategories["SUPER_DEFORMED"] = "super_deformed";
    EKitsuMangaCategories["DETECTIVE"] = "detective";
    EKitsuMangaCategories["DRAMA"] = "drama";
    EKitsuMangaCategories["ANGEL"] = "angel";
    EKitsuMangaCategories["CONTEMPORARY_FANTASY"] = "contemporary_fantasy";
    EKitsuMangaCategories["DARK_FANTASY"] = "dark_fantasy";
    EKitsuMangaCategories["DEITY"] = "deity";
    EKitsuMangaCategories["DEMON"] = "demon";
    EKitsuMangaCategories["DRAGON"] = "dragon";
    EKitsuMangaCategories["ELF"] = "elf";
    EKitsuMangaCategories["HIGH_FANTASY"] = "high_fantasy";
    EKitsuMangaCategories["MAGIC"] = "magic";
    EKitsuMangaCategories["MERMAID"] = "mermaid";
    EKitsuMangaCategories["GHOST"] = "ghost";
    EKitsuMangaCategories["HAREM"] = "harem";
    EKitsuMangaCategories["HENSHIN"] = "henshin";
    EKitsuMangaCategories["HORROR"] = "horror";
    EKitsuMangaCategories["MAGICAL_GIRL"] = "magical_girl";
    EKitsuMangaCategories["MYSTERY"] = "mystery";
    EKitsuMangaCategories["PARASITE"] = "parasite";
    EKitsuMangaCategories["DEMENTIA"] = "dementia";
    EKitsuMangaCategories["PSYCHOLOGICAL"] = "psychological";
    EKitsuMangaCategories["LOVE_POLYGON"] = "love_polygon";
    EKitsuMangaCategories["SHOUJO_AI"] = "shoujo_ai";
    EKitsuMangaCategories["SHOUNEN_AI"] = "shounen_ai";
    EKitsuMangaCategories["SLOW_WHEN_IT_COMES_TO_LOVE"] = "slow_when_it_comes_to_love";
    EKitsuMangaCategories["SUDDEN_GIRLFRIEND_APPEARANCE"] = "sudden_girlfriend_appearance";
    EKitsuMangaCategories["UNREQUITED_LOVE"] = "unrequited_love";
    EKitsuMangaCategories["SCIENCE_FICTION"] = "science_fiction";
    EKitsuMangaCategories["HUMANOID_ALIEN"] = "humanoid_alien";
    EKitsuMangaCategories["CYBERPUNK"] = "cyberpunk";
    EKitsuMangaCategories["CYBORG"] = "cyborg";
    EKitsuMangaCategories["GENETIC_MODIFICATION"] = "genetic_modification";
    EKitsuMangaCategories["POWER_SUIT"] = "power_suit";
    EKitsuMangaCategories["ANDROID"] = "android";
    EKitsuMangaCategories["PILOTED_ROBOT"] = "piloted_robot";
    EKitsuMangaCategories["ROBOT_HELPER"] = "robot_helper";
    EKitsuMangaCategories["SUPER_ROBOT"] = "super_robot";
    EKitsuMangaCategories["TRANSFORMING_CRAFT"] = "transforming_craft";
    EKitsuMangaCategories["SPACE_OPERA"] = "space_opera";
    EKitsuMangaCategories["SPACE_TRAVEL"] = "space_travel";
    EKitsuMangaCategories["STEAMPUNK"] = "steampunk";
    EKitsuMangaCategories["TIME_TRAVEL"] = "time_travel";
    EKitsuMangaCategories["SUPERHERO"] = "superhero";
    EKitsuMangaCategories["SUPERNATURAL"] = "supernatural";
    EKitsuMangaCategories["THRILLER"] = "thriller";
    EKitsuMangaCategories["VAMPIRE"] = "vampire";
    EKitsuMangaCategories["VIRTUAL_REALITY"] = "virtual_reality";
    EKitsuMangaCategories["ZOMBIE"] = "zombie";
    EKitsuMangaCategories["COUNTRYSIDE"] = "countryside";
    EKitsuMangaCategories["DESERT"] = "desert";
    EKitsuMangaCategories["AFRICA"] = "africa";
    EKitsuMangaCategories["UNITED_STATES"] = "united_states";
    EKitsuMangaCategories["NEW_YORK"] = "new_york";
    EKitsuMangaCategories["AMERICAS"] = "americas";
    EKitsuMangaCategories["EARTH"] = "earth";
    EKitsuMangaCategories["SUPER_POWER"] = "super_power";
    EKitsuMangaCategories["MECHA"] = "mecha";
    EKitsuMangaCategories["ROBOT"] = "robot";
    EKitsuMangaCategories["ALIEN"] = "alien";
    EKitsuMangaCategories["ROMANCE"] = "romance";
    EKitsuMangaCategories["ASIA"] = "asia";
    EKitsuMangaCategories["CHINA"] = "china";
    EKitsuMangaCategories["JAPAN"] = "japan";
    EKitsuMangaCategories["KYOTO"] = "kyoto";
    EKitsuMangaCategories["TOKYO"] = "tokyo";
    EKitsuMangaCategories["KOREA"] = "korea";
    EKitsuMangaCategories["FRANCE"] = "france";
    EKitsuMangaCategories["EUROPE"] = "europe";
    EKitsuMangaCategories["ITALY"] = "italy";
    EKitsuMangaCategories["GERMANY"] = "germany";
    EKitsuMangaCategories["RUSSIA"] = "russia";
    EKitsuMangaCategories["UNITED_KINGDOM"] = "united_kingdom";
    EKitsuMangaCategories["MIDDLE_EAST"] = "middle_east";
    EKitsuMangaCategories["FANTASY_WORLD"] = "fantasy_world";
    EKitsuMangaCategories["FLOATING_ISLAND"] = "floating_island";
    EKitsuMangaCategories["FUTURE"] = "future";
    EKitsuMangaCategories["ISEKAI"] = "isekai";
    EKitsuMangaCategories["ISLAND"] = "island";
    EKitsuMangaCategories["PARALLEL_UNIVERSE"] = "parallel_universe";
    EKitsuMangaCategories["ALTERNATIVE_PAST"] = "alternative_past";
    EKitsuMangaCategories["BAKUMATSU_MEIJI_PERIOD"] = "bakumatsu_meiji_period";
    EKitsuMangaCategories["HEIAN_PERIOD"] = "heian_period";
    EKitsuMangaCategories["SENGOKU_PERIOD"] = "sengoku_period";
    EKitsuMangaCategories["THREE_KINGDOMS"] = "three_kingdoms";
    EKitsuMangaCategories["TOKUGAWA_PERIOD"] = "tokugawa_period";
    EKitsuMangaCategories["VICTORIAN_PERIOD"] = "victorian_period";
    EKitsuMangaCategories["WORLD_WAR_II"] = "world_war_ii";
    EKitsuMangaCategories["HISTORICAL"] = "historical";
    EKitsuMangaCategories["PAST"] = "past";
    EKitsuMangaCategories["PRESENT"] = "present";
    EKitsuMangaCategories["ALTERNATIVE_PRESENT"] = "alternative_present";
    EKitsuMangaCategories["OTHER_PLANET"] = "other_planet";
    EKitsuMangaCategories["SHIPBOARD"] = "shipboard";
    EKitsuMangaCategories["MARS"] = "mars";
    EKitsuMangaCategories["SPACE"] = "space";
    EKitsuMangaCategories["SUMMER"] = "summer";
    EKitsuMangaCategories["JOSEI"] = "josei";
    EKitsuMangaCategories["KIDS"] = "kids";
    EKitsuMangaCategories["SEINEN"] = "seinen";
    EKitsuMangaCategories["SHOUJO"] = "shoujo";
    EKitsuMangaCategories["SHOUNEN"] = "shounen";
    EKitsuMangaCategories["ANTI_WAR"] = "anti_war";
    EKitsuMangaCategories["COMING_OF_AGE"] = "coming_of_age";
    EKitsuMangaCategories["CONSPIRACY"] = "conspiracy";
    EKitsuMangaCategories["COOKING"] = "cooking";
    EKitsuMangaCategories["ASSASSIN"] = "assassin";
    EKitsuMangaCategories["BOUNTY_HUNTER"] = "bounty_hunter";
    EKitsuMangaCategories["MAFIA"] = "mafia";
    EKitsuMangaCategories["PIRATE"] = "pirate";
    EKitsuMangaCategories["THIEVERY"] = "thievery";
    EKitsuMangaCategories["EPIDEMIC"] = "epidemic";
    EKitsuMangaCategories["POST_APOCALYPSE"] = "post_apocalypse";
    EKitsuMangaCategories["WAR"] = "war";
    EKitsuMangaCategories["FAMILY"] = "family";
    EKitsuMangaCategories["FRIENDSHIP"] = "friendship";
    EKitsuMangaCategories["GENDER_BENDER"] = "gender_bender";
    EKitsuMangaCategories["COPS"] = "cops";
    EKitsuMangaCategories["SHINSENGUMI"] = "shinsengumi";
    EKitsuMangaCategories["AIR_FORCE"] = "air_force";
    EKitsuMangaCategories["FEUDAL_WARFARE"] = "feudal_warfare";
    EKitsuMangaCategories["NAVY"] = "navy";
    EKitsuMangaCategories["MILITARY"] = "military";
    EKitsuMangaCategories["SPECIAL_SQUADS"] = "special_squads";
    EKitsuMangaCategories["LAW_AND_ORDER"] = "law_and_order";
    EKitsuMangaCategories["DISASTER"] = "disaster";
    EKitsuMangaCategories["NETORARE"] = "netorare";
    EKitsuMangaCategories["PARENTAL_ABANDONMENT"] = "parental_abandonment";
    EKitsuMangaCategories["POLITICS"] = "politics";
    EKitsuMangaCategories["PROXY_BATTLES"] = "proxy_battles";
    EKitsuMangaCategories["BUDDHISM"] = "buddhism";
    EKitsuMangaCategories["REVENGE"] = "revenge";
    EKitsuMangaCategories["ROTTEN_WORLD"] = "rotten_world";
    EKitsuMangaCategories["DYSTOPIA"] = "dystopia";
    EKitsuMangaCategories["SCHOOL_LIFE"] = "school_life";
    EKitsuMangaCategories["ALL_GIRLS_SCHOOL"] = "all_girls_school";
    EKitsuMangaCategories["DELINQUENT"] = "delinquent";
    EKitsuMangaCategories["ELEMENTARY_SCHOOL"] = "elementary_school";
    EKitsuMangaCategories["HIGH_SCHOOL"] = "high_school";
    EKitsuMangaCategories["MIDDLE_SCHOOL"] = "middle_school";
    EKitsuMangaCategories["SCHOOL_CLUBS"] = "school_clubs";
    EKitsuMangaCategories["STUDENT_GOVERNMENT"] = "student_government";
    EKitsuMangaCategories["UNIVERSITY"] = "university";
    EKitsuMangaCategories["RELIGION"] = "religion";
    EKitsuMangaCategories["SLAVERY"] = "slavery";
    EKitsuMangaCategories["SLICE_OF_LIFE"] = "slice_of_life";
    EKitsuMangaCategories["WORKING_LIFE"] = "working_life";
    EKitsuMangaCategories["BASEBALL"] = "baseball";
    EKitsuMangaCategories["BASKETBALL"] = "basketball";
    EKitsuMangaCategories["SPORTS"] = "sports";
    EKitsuMangaCategories["CARD_GAMES"] = "card_games";
    EKitsuMangaCategories["COMBAT"] = "combat";
    EKitsuMangaCategories["CYCLING"] = "cycling";
    EKitsuMangaCategories["BOXING"] = "boxing";
    EKitsuMangaCategories["WRESTLING"] = "wrestling";
    EKitsuMangaCategories["MOTORSPORT"] = "motorsport";
    EKitsuMangaCategories["DRIFTING"] = "drifting";
    EKitsuMangaCategories["FORMULA_RACING"] = "formula_racing";
    EKitsuMangaCategories["STREET_RACING"] = "street_racing";
    EKitsuMangaCategories["SOCCER"] = "soccer";
    EKitsuMangaCategories["TENNIS"] = "tennis";
    EKitsuMangaCategories["VOLLEYBALL"] = "volleyball";
    EKitsuMangaCategories["IDOL"] = "idol";
    EKitsuMangaCategories["MUSICAL_BAND"] = "musical_band";
    EKitsuMangaCategories["PERFORMANCE"] = "performance";
    EKitsuMangaCategories["THE_ARTS"] = "the_arts";
    EKitsuMangaCategories["FANTASY"] = "fantasy";
    EKitsuMangaCategories["ECCHI"] = "ecchi";
    EKitsuMangaCategories["HUMAN_ENHANCEMENT"] = "human_enhancement";
    EKitsuMangaCategories["CRIME"] = "crime";
    EKitsuMangaCategories["MUSIC"] = "music";
})(EKitsuMangaCategories || (EKitsuMangaCategories = {}));
/**
 * @enum
 * @description This enum contain all of anime categories that use Kitsu.app
 * @since 1.3.0
 */
var EKitsuAnimeCategories;
(function (EKitsuAnimeCategories) {
    EKitsuAnimeCategories["VIOLENCE"] = "violence";
    EKitsuAnimeCategories["PLOT_CONTINUITY"] = "plot_continuity";
    EKitsuAnimeCategories["STEREOTYPES"] = "stereotypes";
    EKitsuAnimeCategories["TONE_CHANGES"] = "tone_changes";
    EKitsuAnimeCategories["ACTION"] = "action";
    EKitsuAnimeCategories["BATTLE_ROYALE"] = "battle_royale";
    EKitsuAnimeCategories["GUNFIGHTS"] = "gunfights";
    EKitsuAnimeCategories["MARTIAL_ARTS"] = "martial_arts";
    EKitsuAnimeCategories["NINJA"] = "ninja";
    EKitsuAnimeCategories["SPACE_BATTLES"] = "space_battles";
    EKitsuAnimeCategories["SWORDPLAY"] = "swordplay";
    EKitsuAnimeCategories["SAMURAI"] = "samurai";
    EKitsuAnimeCategories["ADVENTURE"] = "adventure";
    EKitsuAnimeCategories["ANGST"] = "angst";
    EKitsuAnimeCategories["ANIME_INFLUENCED"] = "anime_influenced";
    EKitsuAnimeCategories["ANTHROPOMORPHISM"] = "anthropomorphism";
    EKitsuAnimeCategories["BLACKMAIL"] = "blackmail";
    EKitsuAnimeCategories["COMEDY"] = "comedy";
    EKitsuAnimeCategories["ABSURDIST_HUMOUR"] = "absurdist_humour";
    EKitsuAnimeCategories["BREAKING_THE_FOURTH_WALL"] = "breaking_the_fourth_wall";
    EKitsuAnimeCategories["PARODY"] = "parody";
    EKitsuAnimeCategories["SATIRE"] = "satire";
    EKitsuAnimeCategories["SUPER_DEFORMED"] = "super_deformed";
    EKitsuAnimeCategories["VIOLENT_RETRIBUTION_FOR_ACCIDENTAL_INFRINGEMENT"] = "violent_retribution_for_accidental_infringement";
    EKitsuAnimeCategories["DETECTIVE"] = "detective";
    EKitsuAnimeCategories["DRAMA"] = "drama";
    EKitsuAnimeCategories["FANTASY"] = "fantasy";
    EKitsuAnimeCategories["ANGEL"] = "angel";
    EKitsuAnimeCategories["CONTEMPORARY_FANTASY"] = "contemporary_fantasy";
    EKitsuAnimeCategories["DARK_FANTASY"] = "dark_fantasy";
    EKitsuAnimeCategories["DEITY"] = "deity";
    EKitsuAnimeCategories["DEMON"] = "demon";
    EKitsuAnimeCategories["DRAGON"] = "dragon";
    EKitsuAnimeCategories["ELF"] = "elf";
    EKitsuAnimeCategories["HIGH_FANTASY"] = "high_fantasy";
    EKitsuAnimeCategories["MAGIC"] = "magic";
    EKitsuAnimeCategories["MERMAID"] = "mermaid";
    EKitsuAnimeCategories["GHOST"] = "ghost";
    EKitsuAnimeCategories["HAREM"] = "harem";
    EKitsuAnimeCategories["HENSHIN"] = "henshin";
    EKitsuAnimeCategories["HORROR"] = "horror";
    EKitsuAnimeCategories["MAGICAL_GIRL"] = "magical_girl";
    EKitsuAnimeCategories["MYSTERY"] = "mystery";
    EKitsuAnimeCategories["PARASITE"] = "parasite";
    EKitsuAnimeCategories["DEMENTIA"] = "dementia";
    EKitsuAnimeCategories["LOVE_POLYGON"] = "love_polygon";
    EKitsuAnimeCategories["PSYCHOLOGICAL"] = "psychological";
    EKitsuAnimeCategories["ROMANCE"] = "romance";
    EKitsuAnimeCategories["SHOUJO_AI"] = "shoujo_ai";
    EKitsuAnimeCategories["SHOUNEN_AI"] = "shounen_ai";
    EKitsuAnimeCategories["SLOW_WHEN_IT_COMES_TO_LOVE"] = "slow_when_it_comes_to_love";
    EKitsuAnimeCategories["SUDDEN_GIRLFRIEND_APPEARANCE"] = "sudden_girlfriend_appearance";
    EKitsuAnimeCategories["UNREQUITED_LOVE"] = "unrequited_love";
    EKitsuAnimeCategories["ECCHI"] = "ecchi";
    EKitsuAnimeCategories["SLAPSTICK"] = "slapstick";
    EKitsuAnimeCategories["SCIENCE_FICTION"] = "science_fiction";
    EKitsuAnimeCategories["ALIEN"] = "alien";
    EKitsuAnimeCategories["HUMANOID_ALIEN"] = "humanoid_alien";
    EKitsuAnimeCategories["CYBERPUNK"] = "cyberpunk";
    EKitsuAnimeCategories["HUMAN_ENHANCEMENT"] = "human_enhancement";
    EKitsuAnimeCategories["CYBORG"] = "cyborg";
    EKitsuAnimeCategories["GENETIC_MODIFICATION"] = "genetic_modification";
    EKitsuAnimeCategories["MECHA"] = "mecha";
    EKitsuAnimeCategories["ROBOT"] = "robot";
    EKitsuAnimeCategories["POWER_SUIT"] = "power_suit";
    EKitsuAnimeCategories["SPACE_OPERA"] = "space_opera";
    EKitsuAnimeCategories["SPACE_TRAVEL"] = "space_travel";
    EKitsuAnimeCategories["STEAMPUNK"] = "steampunk";
    EKitsuAnimeCategories["TIME_TRAVEL"] = "time_travel";
    EKitsuAnimeCategories["SUPER_POWER"] = "super_power";
    EKitsuAnimeCategories["SUPERHERO"] = "superhero";
    EKitsuAnimeCategories["SUPERNATURAL"] = "supernatural";
    EKitsuAnimeCategories["THRILLER"] = "thriller";
    EKitsuAnimeCategories["VAMPIRE"] = "vampire";
    EKitsuAnimeCategories["ZOMBIE"] = "zombie";
    EKitsuAnimeCategories["VIRTUAL_REALITY"] = "virtual_reality";
    EKitsuAnimeCategories["COUNTRYSIDE"] = "countryside";
    EKitsuAnimeCategories["DESERT"] = "desert";
    EKitsuAnimeCategories["EARTH"] = "earth";
    EKitsuAnimeCategories["AFRICA"] = "africa";
    EKitsuAnimeCategories["AMERICAS"] = "americas";
    EKitsuAnimeCategories["UNITED_STATES"] = "united_states";
    EKitsuAnimeCategories["NEW_YORK"] = "new_york";
    EKitsuAnimeCategories["CHINA"] = "china";
    EKitsuAnimeCategories["JAPAN"] = "japan";
    EKitsuAnimeCategories["KYOTO"] = "kyoto";
    EKitsuAnimeCategories["TOKYO"] = "tokyo";
    EKitsuAnimeCategories["EUROPE"] = "europe";
    EKitsuAnimeCategories["KOREA"] = "korea";
    EKitsuAnimeCategories["MIDDLE_EAST"] = "middle_east";
    EKitsuAnimeCategories["FRANCE"] = "france";
    EKitsuAnimeCategories["GERMANY"] = "germany";
    EKitsuAnimeCategories["ITALY"] = "italy";
    EKitsuAnimeCategories["RUSSIA"] = "russia";
    EKitsuAnimeCategories["UNITED_KINGDOM"] = "united_kingdom";
    EKitsuAnimeCategories["FLOATING_ISLAND"] = "floating_island";
    EKitsuAnimeCategories["FANTASY_WORLD"] = "fantasy_world";
    EKitsuAnimeCategories["FUTURE"] = "future";
    EKitsuAnimeCategories["ISEKAI"] = "isekai";
    EKitsuAnimeCategories["ISLAND"] = "island";
    EKitsuAnimeCategories["PARALLEL_UNIVERSE"] = "parallel_universe";
    EKitsuAnimeCategories["PAST"] = "past";
    EKitsuAnimeCategories["ALTERNATIVE_PAST"] = "alternative_past";
    EKitsuAnimeCategories["BAKUMATSU_MEIJI_PERIOD"] = "bakumatsu_meiji_period";
    EKitsuAnimeCategories["HEIAN_PERIOD"] = "heian_period";
    EKitsuAnimeCategories["SENGOKU_PERIOD"] = "sengoku_period";
    EKitsuAnimeCategories["THREE_KINGDOMS"] = "three_kingdoms";
    EKitsuAnimeCategories["TOKUGAWA_PERIOD"] = "tokugawa_period";
    EKitsuAnimeCategories["VICTORIAN_PERIOD"] = "victorian_period";
    EKitsuAnimeCategories["WORLD_WAR_II"] = "world_war_ii";
    EKitsuAnimeCategories["HISTORICAL"] = "historical";
    EKitsuAnimeCategories["PRESENT"] = "present";
    EKitsuAnimeCategories["ALTERNATIVE_PRESENT"] = "alternative_present";
    EKitsuAnimeCategories["SPACE"] = "space";
    EKitsuAnimeCategories["OTHER_PLANET"] = "other_planet";
    EKitsuAnimeCategories["MARS"] = "mars";
    EKitsuAnimeCategories["SHIPBOARD"] = "shipboard";
    EKitsuAnimeCategories["SUMMER"] = "summer";
    EKitsuAnimeCategories["JOSEI"] = "josei";
    EKitsuAnimeCategories["KIDS"] = "kids";
    EKitsuAnimeCategories["SEINEN"] = "seinen";
    EKitsuAnimeCategories["SHOUJO"] = "shoujo";
    EKitsuAnimeCategories["SHOUNEN"] = "shounen";
    EKitsuAnimeCategories["ANTI_WAR"] = "anti_war";
    EKitsuAnimeCategories["COMING_OF_AGE"] = "coming_of_age";
    EKitsuAnimeCategories["CONSPIRACY"] = "conspiracy";
    EKitsuAnimeCategories["COOKING"] = "cooking";
    EKitsuAnimeCategories["CRIME"] = "crime";
    EKitsuAnimeCategories["ASSASSIN"] = "assassin";
    EKitsuAnimeCategories["BOUNTY_HUNTER"] = "bounty_hunter";
    EKitsuAnimeCategories["MAFIA"] = "mafia";
    EKitsuAnimeCategories["PIRATE"] = "pirate";
    EKitsuAnimeCategories["THIEVERY"] = "thievery";
    EKitsuAnimeCategories["DISASTER"] = "disaster";
    EKitsuAnimeCategories["EPIDEMIC"] = "epidemic";
    EKitsuAnimeCategories["POST_APOCALYPSE"] = "post_apocalypse";
    EKitsuAnimeCategories["WAR"] = "war";
    EKitsuAnimeCategories["FAMILY"] = "family";
    EKitsuAnimeCategories["FRIENDSHIP"] = "friendship";
    EKitsuAnimeCategories["GENDER_BENDER"] = "gender_bender";
    EKitsuAnimeCategories["LAW_AND_ORDER"] = "law_and_order";
    EKitsuAnimeCategories["COPS"] = "cops";
    EKitsuAnimeCategories["SPECIAL_SQUADS"] = "special_squads";
    EKitsuAnimeCategories["SHINSENGUMI"] = "shinsengumi";
    EKitsuAnimeCategories["MILITARY"] = "military";
    EKitsuAnimeCategories["AIR_FORCE"] = "air_force";
    EKitsuAnimeCategories["FEUDAL_WARFARE"] = "feudal_warfare";
    EKitsuAnimeCategories["NAVY"] = "navy";
    EKitsuAnimeCategories["NETORARE"] = "netorare";
    EKitsuAnimeCategories["PARENTAL_ABANDONMENT"] = "parental_abandonment";
    EKitsuAnimeCategories["POLITICS"] = "politics";
    EKitsuAnimeCategories["PROXY_BATTLES"] = "proxy_battles";
    EKitsuAnimeCategories["RELIGION"] = "religion";
    EKitsuAnimeCategories["BUDDHISM"] = "buddhism";
    EKitsuAnimeCategories["REVENGE"] = "revenge";
    EKitsuAnimeCategories["DYSTOPIA"] = "dystopia";
    EKitsuAnimeCategories["SCHOOL_LIFE"] = "school_life";
    EKitsuAnimeCategories["MUSIC"] = "music";
})(EKitsuAnimeCategories || (EKitsuAnimeCategories = {}));
/**
 * @enum
 * @description This enum contain the basic seasons of the year.
 * @since 1.3.0
 */
var EKitsuSeason;
(function (EKitsuSeason) {
    EKitsuSeason["Spring"] = "spring";
    EKitsuSeason["Summer"] = "summer";
    EKitsuSeason["Fall"] = "fall";
    EKitsuSeason["Autumn"] = "fall";
    EKitsuSeason["Winter"] = "winter";
})(EKitsuSeason || (EKitsuSeason = {}));
/**
 * @enum
 * @description This enum contain all of streamers that is available.
 * @since 1.3.0
 */
var EKitsuAnimeStreamers;
(function (EKitsuAnimeStreamers) {
    EKitsuAnimeStreamers["Crunchyroll"] = "Crunchyroll";
    EKitsuAnimeStreamers["Hulu"] = "Hulu";
    EKitsuAnimeStreamers["Funanimation"] = "Funanimation";
    EKitsuAnimeStreamers["ConTV"] = "CONtv";
    EKitsuAnimeStreamers["Netflix"] = "Netflix";
    EKitsuAnimeStreamers["Hidive"] = "HIDIVE";
    EKitsuAnimeStreamers["TubiTV"] = "TubiTV";
    EKitsuAnimeStreamers["Amazon"] = "Amazon";
    EKitsuAnimeStreamers["Youtube"] = "Youtube";
    EKitsuAnimeStreamers["AnimeLab"] = "AnimeLab";
    EKitsuAnimeStreamers["VRV"] = "VRV";
})(EKitsuAnimeStreamers || (EKitsuAnimeStreamers = {}));
/**
 * @enum
 * @description This enum contain all of age rating categories that use Kitsu.app
 * @since 1.3.0
 */
var EKitsuAnimeAgeRating;
(function (EKitsuAnimeAgeRating) {
    EKitsuAnimeAgeRating["G"] = "G";
    EKitsuAnimeAgeRating["PG"] = "PG";
    EKitsuAnimeAgeRating["R"] = "R";
    EKitsuAnimeAgeRating["ParentalGuidanceSuggested"] = "PG";
    EKitsuAnimeAgeRating["GeneralAudience"] = "G";
    EKitsuAnimeAgeRating["Restricted"] = "R";
})(EKitsuAnimeAgeRating || (EKitsuAnimeAgeRating = {}));
/**
 * @enum
 * @description This enum contain all of anime sub types categories that use Kitsu.app
 * @since 1.3.0
 */
var EKitsuAnimeSubTypes;
(function (EKitsuAnimeSubTypes) {
    EKitsuAnimeSubTypes["ONA"] = "ONA";
    EKitsuAnimeSubTypes["OVA"] = "OVA";
    EKitsuAnimeSubTypes["TV"] = "TV";
    EKitsuAnimeSubTypes["Movie"] = "movie";
    EKitsuAnimeSubTypes["Music"] = "music";
    EKitsuAnimeSubTypes["Special"] = "Special";
})(EKitsuAnimeSubTypes || (EKitsuAnimeSubTypes = {}));
/**
 * @enum
 * @description This enum contain all of manga sub types categories that use Kitsu.app
 * @since 1.3.0
 */
var EKitsuMangaSubTypes;
(function (EKitsuMangaSubTypes) {
    EKitsuMangaSubTypes["manga"] = "manga";
    EKitsuMangaSubTypes["Roman"] = "roman";
    EKitsuMangaSubTypes["Manhua"] = "manhua";
    EKitsuMangaSubTypes["Manhwa"] = "manhwa";
    EKitsuMangaSubTypes["OneShot"] = "oneshot";
    EKitsuMangaSubTypes["Doujin"] = "doujin";
    EKitsuMangaSubTypes["AmeriManga"] = "oel";
})(EKitsuMangaSubTypes || (EKitsuMangaSubTypes = {}));
export { EKitsuAnimeAgeRating, EKitsuAnimeCategories, EKitsuAnimeStreamers, EKitsuAnimeSubTypes, EKitsuMangaCategories, EKitsuMangaSubTypes, EKitsuSeason, };
